symplyphysics/laws/electricity/transmission_lines/transmission_matrix_of_pi_type_circuit.py

```diff
@@ -0,0 +1,99 @@
+## Z1 - first impedance,
+## Z2 - second impedance,
+## Z3 - third impedance.
+
+parameter_voltage_to_voltage = Symbol("parameter_voltage_to_voltage", dimensionless)
+parameter_impedance = Symbol("parameter_impedance", units.impedance)
+parameter_conductance = Symbol("parameter_conductance", units.conductance)
+parameter_current_to_current = Symbol("parameter_current_to_current", dimensionless)
+
+first_impedance = Symbol("first_impedance", units.impedance)
+second_impedance = Symbol("second_impedance", units.impedance)
+third_impedance = Symbol("third_impedance", units.impedance)
+
+
+law = Eq(Matrix([[parameter_voltage_to_voltage, parameter_impedance], [parameter_conductance, parameter_current_to_current]]),
+         Matrix([[1 + third_impedance / second_impedance, third_impedance], [(1 / first_impedance) + (1 / second_impedance) + third_impedance / (first_impedance * second_impedance), 1 + third_impedance / first_impedance]]))
+
+# This law might be derived via "transmission_matrix_for_a_series_load_in_line" law and
+# "transmission_matrix_for_a_parallel_load_in_line" law.
+
+parallel_law_applied_1 = parallel_law.law.subs({
+    parallel_law.load_impedance: first_impedance,
+})
+matrix_derived_1 = solve(parallel_law_applied_1, [parallel_law.parameter_voltage_to_voltage, parallel_law.parameter_impedance, parallel_law.parameter_conductance, parallel_law.parameter_current_to_current], dict=True)[0]
+matrix_derived_1 = Matrix([[matrix_derived_1[parallel_law.parameter_voltage_to_voltage], matrix_derived_1[parallel_law.parameter_impedance]], [matrix_derived_1[parallel_law.parameter_conductance], matrix_derived_1[parallel_law.parameter_current_to_current]]])
+
+series_law_applied = series_law.law.subs({
+    series_law.load_impedance: third_impedance,
+})
+matrix_derived_3 = solve(series_law_applied, [series_law.parameter_voltage_to_voltage, series_law.parameter_impedance, series_law.parameter_conductance, series_law.parameter_current_to_current], dict=True)[0]
+matrix_derived_3 = Matrix([[matrix_derived_3[series_law.parameter_voltage_to_voltage], matrix_derived_3[series_law.parameter_impedance]], [matrix_derived_3[series_law.parameter_conductance], matrix_derived_3[series_law.parameter_current_to_current]]])
+
+parallel_law_applied_2 = parallel_law.law.subs({
+    parallel_law.load_impedance: second_impedance,
+})
+matrix_derived_2 = solve(parallel_law_applied_2, [parallel_law.parameter_voltage_to_voltage, parallel_law.parameter_impedance, parallel_law.parameter_conductance, parallel_law.parameter_current_to_current], dict=True)[0]
+matrix_derived_2 = Matrix([[matrix_derived_2[parallel_law.parameter_voltage_to_voltage], matrix_derived_2[parallel_law.parameter_impedance]], [matrix_derived_2[parallel_law.parameter_conductance], matrix_derived_2[parallel_law.parameter_current_to_current]]])
+
+# When cascading elements, the final transfer matrix will be equal to the product of the matrices of each element.
```

Comment:
Is it possible to make a law from it?

Reply:
> Is it possible to make a law from it?

Yes, probably. But i need a function or class that can multiply matrices. I don't find a similar function in Sympy.

Reply:
I think Product should suffice. It is like Sum, but for multiplication.

Reply:
> I think Product should suffice. It is like Sum, but for multiplication.


It's not working out yet. The code does not work because an error appears on the line with "Product":
```
ValueError
specify dummy variables for Matrix([[parameters_voltage_to_voltage1, parameters_impedance1], [parameters_conductance1, parameters_current_to_current1]])
```
My code
```python
parameter_voltage_to_voltage = Symbol("parameter_voltage_to_voltage", dimensionless)
parameter_impedance = Symbol("parameter_impedance", units.impedance)
parameter_conductance = Symbol("parameter_conductance", units.conductance)
parameter_current_to_current = Symbol("parameter_current_to_current", dimensionless)

parameters_voltage_to_voltage = SymbolIndexed("parameters_voltage_to_voltage", dimensionless)
parameters_impedance = SymbolIndexed("parameters_impedance", units.impedance)
parameters_conductance = SymbolIndexed("parameters_conductance", units.conductance)
parameters_current_to_current = SymbolIndexed("parameters_current_to_current", dimensionless)

law = Eq(Matrix([[parameter_voltage_to_voltage, parameter_impedance], [parameter_conductance, parameter_current_to_current]]),
         Product(Matrix([[parameters_voltage_to_voltage, parameters_impedance], [parameters_conductance, parameters_current_to_current]])))

def calculate_transmission_matrix(parameters_: tuple[tuple[float, Quantity], tuple[Quantity, float]]) -> tuple[tuple[float, Quantity], tuple[Quantity, float]]:
    local_index = Idx("index_local", (1, len(parameters_)))
    matrix_law = law.subs(global_index, local_index)
    matrix_law = matrix_law.doit()
    solved = solve(matrix_law, [parameter_voltage_to_voltage, parameter_impedance, parameter_conductance, parameter_current_to_current], dict=True)[0]
    result_A = solved[parameter_voltage_to_voltage]
    result_B = solved[parameter_impedance]
    result_C = solved[parameter_conductance]
    result_D = solved[parameter_current_to_current]
    for i, v in enumerate(parameters_):
        result_A = result_A.subs(parameters_voltage_to_voltage[i + 1], v[0][0])
        result_B = result_B.subs(parameters_impedance[i + 1], v[0][1])
        result_C = result_C.subs(parameters_conductance[i + 1], v[1][0])
        result_D = result_D.subs(parameters_current_to_current[i + 1], v[1][1])
```

At the moment I don't know how to solve this problem

Reply:
You do not need to solve it. Product is not ready for use. Fix other issues and we're done.